engine/remotecache/cache.go

```diff
@@ -69,7 +81,7 @@ func ResolveCacheImporterFunc(sm *session.Manager, cs content.Store, hosts docke
 		case "gha":
 			impl, desc, err = gha.ResolveCacheImporterFunc()(ctx, g, attrs)
 		case "s3":
-			impl, desc, err = s3remotecache.ResolveCacheImporterFunc()(ctx, g, attrs)
+			impl, desc, err = s3Importer(ctx, g, attrs)
```

Comment:
Do you know how often this will be called by BuildKit? Has me wondering if we should keep a "cache" of the keys + manifests, so we don't have to download them every time this gets called within a session.

Reply:
It will be called once per session, so once per `dagger.Connect`. Right now all the manifests for each import in the pool will be re-downloaded at that time. I suspect there are other bottlenecks before this one, but yeah we could have something like an LRU cache which would reduce many imports to just be a HEAD instead of a full GET. Might even be a good candidate for upstreaming since it's not necessarily specific to our use case.

Reply:
> It will be called once per session, so once per `dagger.Connect`. Right now all the manifests for each import in the pool will be re-downloaded at that time. I suspect there are other bottlenecks before this one, but yeah we could have something like an LRU cache which would reduce many imports to just be a HEAD instead of a full GET. Might even be a good candidate for upstreaming since it's not necessarily specific to our use case.

I rather like the idea of having a global LRU cache like you describe. That could definitely improve some of the performance of getting manifests, etc.

That said, I agree about other bottlenecks likely being an issues sooner. :)